### PR TITLE
[.NET] Dutch DateTime Holiday support

### PR DESCRIPTION
**Pull request by LionbridgeCS**
All test cases pass for the Extractor and the Parser

### CODE REVIEW

**Comment by LionbridgeCS** [comment]
@tellarin, @iMicknl, this is ready for review.

**Comment by LionbridgeCS** [comment]
@iMicknl: The parser test has a couple of cases with volgend: "Ik ga terug op Nieuwjaarsdag volgend jaar" and "Ik ga terug op Yuandan volgend jaar". are more cases needed?

File: .NET/Microsoft.Recognizers.Definitions.Common/Dutch/DateTimeDefinitions.cs [hunk 1]
```diff
@@ -652,7 +653,7 @@ public static class DateTimeDefinitions
             { @"christmaseve", new string[] { @"kerstavond" } },
             { @"columbus", new string[] { @"columbusday" } },
             { @"thanksgiving", new string[] { @"thanksgivingday", @"thanksgiving", @"dankzeggingsdag" } },
-            { @"martinlutherking", new string[] { @"martinlutherkingday", @"martinlutherkingjrday" } },
+            { @"martinlutherking", new string[] { @"martinlutherkingday", @"martinlutherkingjrday", @"martinlutherkingdag", @"martin luther king dag", @"mlkdag" } },
```
Comment (tellarin) [inline .NET/Microsoft.Recognizers.Definitions.Common/Dutch/DateTimeDefinitions.cs]: Values with spaces should not be needed in the dictionary as there is a simple normalization before querying, right?

File: .NET/Microsoft.Recognizers.Definitions.Common/Dutch/DateTimeDefinitions.cs [hunk 2]
```diff
@@ -652,7 +653,7 @@ public static class DateTimeDefinitions
             { @"christmaseve", new string[] { @"kerstavond" } },
             { @"columbus", new string[] { @"columbusday" } },
             { @"thanksgiving", new string[] { @"thanksgivingday", @"thanksgiving", @"dankzeggingsdag" } },
-            { @"martinlutherking", new string[] { @"martinlutherkingday", @"martinlutherkingjrday" } },
+            { @"martinlutherking", new string[] { @"martinlutherkingday", @"martinlutherkingjrday", @"martinlutherkingdag", @"martin luther king dag", @"mlkdag" } },
             { @"washingtonsbirthday", new string[] { @"washingtonsbirthday", @"washingtonbirthday" } },
             { @"canberra", new string[] { @"canberraday" } },
             { @"memorial", new string[] { @"memorialday", @"memorial day" } },
```
Comment (tellarin) [inline .NET/Microsoft.Recognizers.Definitions.Common/Dutch/DateTimeDefinitions.cs]: Same

File: .NET/Microsoft.Recognizers.Text.DateTime/Dutch/Parsers/DutchHolidayParserConfiguration.cs [hunk 3]
```diff
@@ -1,100 +1,101 @@
...
-                { "prinsjesdag", Prinsjesdag },
-                { "eerstekerstdag", ChristmasDay },
-                { "tweedekerstdag", SecondChristmasDay },
-                { "bevrijdingsdag", Bevrijdingsdag },
-                { "dodenherdenking", Dodenherdenking },
-                { "dagvandearbeid", Dagvandearbeid },
-            };
-        }
-
+﻿using System;
```
Comment (tellarin) [inline .NET/Microsoft.Recognizers.Text.DateTime/Dutch/Parsers/DutchHolidayParserConfiguration.cs]: Are these line-break changes really necessary to be consistent with other files?
If not, please revert and be careful not to make such changes unless really necessary.

File: .NET/Microsoft.Recognizers.Text.DateTime/Dutch/Parsers/DutchHolidayParserConfiguration.cs [hunk 4]
```diff
@@ -1,100 +1,101 @@
...
+        {
+            this.HolidayRegexList = DutchHolidayExtractorConfiguration.HolidayRegexList;
+            this.HolidayNames = DateTimeDefinitions.HolidayNames.ToImmutableDictionary();
+        }
+
+        public override int GetSwiftYear(string text)
+        {
+            var trimmedText = text.Trim().ToLowerInvariant();
+            var swift = -10;
+            if (trimmedText.StartsWith("volgend") || trimmedText.StartsWith("volgende"))
```
Comment (tellarin) [inline .NET/Microsoft.Recognizers.Text.DateTime/Dutch/Parsers/DutchHolidayParserConfiguration.cs]: Such constants should be hard-coded. Can you move them to YAML?
There are other examples of similar cases in the English YAML that could be used as a base.

File: Patterns/Dutch/Dutch-DateTime.yaml [hunk 5]
```diff
@@ -1035,16 +1040,16 @@ HolidayNames: !dictionary
     whiteloverday: [ whiteloverday ] # TODO or remove?
     loverday: [ loverday ] # TODO or remove?
     groundhougday: [ groundhougday, bosmarmottendag ]
-    inaugurationday: [ inaugurationday, inauguratiedag ] 
+    inaugurationday: [ inaugurationday, inauguratiedag ]
     arborday: [ arborday ] # TODO or remove?
```
Comment (tellarin) [inline Patterns/Dutch/Dutch-DateTime.yaml]: If these holidays marked with TODO are not really mentioned in Dutch, feel free to remove them.

File: .NET/Microsoft.Recognizers.Text.DateTime/Dutch/Parsers/DutchHolidayParserConfiguration.cs (hunk 3 above)
Comment (LionbridgeCS) [reply .NET/Microsoft.Recognizers.Text.DateTime/Dutch/Parsers/DutchHolidayParserConfiguration.cs]: the line ending on this file was Unix style which is inconsistent with other similar files. 
if that is not an issue we can revert this.

File: .NET/Microsoft.Recognizers.Definitions.Common/Dutch/DateTimeDefinitions.cs (hunk 2 above)
Comment (LionbridgeCS) [reply .NET/Microsoft.Recognizers.Definitions.Common/Dutch/DateTimeDefinitions.cs]: noted, will be fixed.

File: .NET/Microsoft.Recognizers.Text.DateTime/Dutch/Parsers/DutchHolidayParserConfiguration.cs (hunk 3 above)
Comment (tellarin) [reply .NET/Microsoft.Recognizers.Text.DateTime/Dutch/Parsers/DutchHolidayParserConfiguration.cs]: If this is now consistent with the rest of the codebase that's fine.

Comment (tellarin) [reply .NET/Microsoft.Recognizers.Text.DateTime/Dutch/Parsers/DutchHolidayParserConfiguration.cs]: If such cases happen in the future, please make a comment in the PR description to clarify. Thanks!

File: .NET/Microsoft.Recognizers.Text.DateTime/Dutch/Parsers/DutchHolidayParserConfiguration.cs (hunk 4 above)
Comment (LionbridgeCS) [reply .NET/Microsoft.Recognizers.Text.DateTime/Dutch/Parsers/DutchHolidayParserConfiguration.cs]: values already in the yaml. changed in the parser to check from the yaml.

File: Patterns/Dutch/Dutch-DateTime.yaml (hunk 5 above)
Comment (LionbridgeCS) [reply Patterns/Dutch/Dutch-DateTime.yaml]: updated.

File: .NET/Microsoft.Recognizers.Definitions.Common/Dutch/DateTimeDefinitions.cs [hunk 6]
```diff
@@ -651,27 +651,20 @@ public static class DateTimeDefinitions
             { @"allsaintsday", new string[] { @"allerheiligen" } },
             { @"allsoulsday", new string[] { @"allerzielen" } },
             { @"christmaseve", new string[] { @"kerstavond" } },
-            { @"columbus", new string[] { @"columbusday" } },
+            { @"columbus", new string[] { @"columbusday", @"columbusday" } },
```
Comment (tellarin) [inline .NET/Microsoft.Recognizers.Definitions.Common/Dutch/DateTimeDefinitions.cs]: There should be no duplicate values in the dictionary.

Comment (LionbridgeCS) [reply .NET/Microsoft.Recognizers.Definitions.Common/Dutch/DateTimeDefinitions.cs]: fixed.

File: .NET/Microsoft.Recognizers.Text.DateTime/Dutch/Parsers/DutchHolidayParserConfiguration.cs (hunk 4 above)
Comment (tellarin) [reply .NET/Microsoft.Recognizers.Text.DateTime/Dutch/Parsers/DutchHolidayParserConfiguration.cs]: These strings ("volgend" and "volgende") should not be hardcoded here. Their values need to be read from the YAML file (similarly to DateTimeDefinitions.HolidayNames). English has examples of such cases.

File: .NET/Microsoft.Recognizers.Text.DateTime/Dutch/Parsers/DutchHolidayParserConfiguration.cs [hunk 7]
```diff
@@ -1,100 +1,113 @@
...
+
+        public Regex NextPrefixRegex { get; }
+
+        public Regex PreviousPrefixRegex { get; }
+
+        public override int GetSwiftYear(string text)
+        {
+            var trimmedText = text.Trim();
+            var swift = -10;
+            if (NextPrefixRegex.IsMatch(trimmedText))
```
Comment (LionbridgeCS) [inline .NET/Microsoft.Recognizers.Text.DateTime/Dutch/Parsers/DutchHolidayParserConfiguration.cs]: @tellarin, we already changed the hard coded to be checked from the yaml

**Pull request merged**